### PR TITLE
Refactor wallpaper view

### PR DESCRIPTION
**Pull request by lenemter**
Significant refactoring of wallpaper view and wallpaper containers. Doesn't fix anything, shouldn't break anything. Targets `gtk4` branch this time

Rework of wallpaper containers inheritance. Unifies the way wallpapers are set, removes useless checks, relies more on GLib.Settings instead of global variables, moves methods closer to their use, renames some methods, small one-time methods were moved into lambdas

### CODE REVIEW

File: src/Widgets/GenericContainer.vala [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        }
+        set {
+            check_revealer.reveal_child = value;
+            if (value) {
+                card_box.set_state_flags (Gtk.StateFlags.CHECKED, false);
+            } else {
+                card_box.unset_state_flags (Gtk.StateFlags.CHECKED);
+            }
+        }
+
```
Comment (jeremypw) [inline src/Widgets/GenericContainer.vala:50]: Extraneous blank line

**Review by jeremypw — COMMENTED** [review]
This seems to work with no regression but I was wondering if the class "UriContainer" is needed since it is never instantiated and, afaict, "ImageContainer" is the only subclass?

**Comment by lenemter** [comment]
@jeremypw I made it this way for easy resolving #296 after GTK 4 port, so xml wallpapers can just subclass UriContainer for not breaking sorting and other stuff.

**Comment by jeremypw** [comment]
> @jeremypw I made it this way for easy resolving #296 after GTK 4 port, so xml wallpapers can just subclass UriContainer for not breaking sorting and other stuff.

Ah, OK.  In that case, it might be an idea to implement a skeleton `XMLContainer` class here or otherwise document what the purpose of `URIContainer` is.

**Pull request merged**